### PR TITLE
engine_rocks: support change rate limit dynamically

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Part of https://github.com/tikv/tikv/issues/6484

### What is changed and how it works?

What's Changed:
This PR is trying to support changing `rate-bytes-per-sec` dynamically. Related to tikv/rust-rocksdb#510

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- Support changing `rate-bytes-per-sec` dynamically

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1938,7 +1938,7 @@ dependencies = [
 [[package]]
 name = "librocksdb_sys"
 version = "0.1.0"
-source = "git+https://github.com/tikv/rust-rocksdb.git#639b47773f9c5f3affddf9f804bf67312a2d0258"
+source = "git+https://github.com/tikv/rust-rocksdb.git#c25b623d0e12c35d8de10f3c2478f2f19a8ac7a3"
```
Comment (yiwu-arbug) [inline Cargo.lock:1941]: update rust-rocksdb here or merge #8137 first?

File: src/config.rs [hunk 2]
```diff
@@ -2885,11 +2900,11 @@ mod tests {
         );
     }
 
-    fn new_engines(cfg: TiKvConfig) -> (RocksEngine, ConfigController) {
+    fn new_engines(cfg: TiKvConfig, db_opt: DBOptions) -> (RocksEngine, ConfigController) {
         let engine = RocksEngine::from_db(Arc::new(
             new_engine_opt(
                 &cfg.storage.data_dir,
-                DBOptions::new(),
+                db_opt,
```
Comment (yiwu-arbug) [inline src/config.rs:2907]: the `db_opt` should be created from `TiKvConfig` instead of passing from parameter?

File: src/config.rs [hunk 3]
```diff
@@ -1326,6 +1331,16 @@ impl ConfigManager for DBConfigManger {
                 }
             }
         }
+
+        let rate_bytes_config = change
+            .drain_filter(|(name, _)| name == "rate_bytes_per_sec")
+            .collect::<Vec<_>>();
```
Comment (Connor1996) [inline src/config.rs:1337]: no need to collect as a vec, maybe just get the first

File: src/config.rs [hunk 4]
```diff
@@ -2911,12 +2926,14 @@ mod tests {
     #[test]
     fn test_change_rocksdb_config() {
         let (mut cfg, _dir) = TiKvConfig::with_tmp().unwrap();
+        let mut init_db_opts = DBOptions::new();
+        init_db_opts.set_ratelimiter(ReadableSize::mb(64).0 as i64);
```
Comment (Connor1996) [inline src/config.rs:2930]: there is a related config in `TiKvConfig`, do not set dboptions directly

**Comment by BusyJay** [comment]
@yiwu-arbug can it be picked to 4.0?

**Comment by yiwu-arbug** [comment]
@Connor1996 are you able to help?

**Pull request merged**